bundles/aliyun/src/main/java/org/apache/gravitino/oss/fs/OSSCredentialProvider.java

```diff
@@ -0,0 +1,132 @@
+/*
+ *  Licensed to the Apache Software Foundation (ASF) under one
+ *  or more contributor license agreements.  See the NOTICE file
+ *  distributed with this work for additional information
+ *  regarding copyright ownership.  The ASF licenses this file
+ *  to you under the Apache License, Version 2.0 (the
+ *  "License"); you may not use this file except in compliance
+ *  with the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing,
+ *  software distributed under the License is distributed on an
+ *  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ *  KIND, either express or implied.  See the License for the
+ *  specific language governing permissions and limitations
+ *  under the License.
+ */
+
+package org.apache.gravitino.oss.fs;
+
+import static org.apache.gravitino.credential.OSSTokenCredential.GRAVITINO_OSS_SESSION_ACCESS_KEY_ID;
+import static org.apache.gravitino.credential.OSSTokenCredential.GRAVITINO_OSS_SESSION_SECRET_ACCESS_KEY;
+import static org.apache.gravitino.credential.OSSTokenCredential.GRAVITINO_OSS_TOKEN;
+
+import com.aliyun.oss.common.auth.BasicCredentials;
+import com.aliyun.oss.common.auth.Credentials;
+import com.aliyun.oss.common.auth.CredentialsProvider;
+import com.aliyun.oss.common.auth.DefaultCredentials;
+import java.net.URI;
+import java.util.Map;
+import org.apache.gravitino.NameIdentifier;
+import org.apache.gravitino.client.GravitinoClient;
+import org.apache.gravitino.credential.Credential;
+import org.apache.gravitino.credential.OSSTokenCredential;
+import org.apache.gravitino.file.Fileset;
+import org.apache.gravitino.file.FilesetCatalog;
+import org.apache.gravitino.filesystem.hadoop.GravitinoVirtualFileSystem;
+import org.apache.gravitino.filesystem.hadoop.GravitinoVirtualFileSystemConfiguration;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.fs.aliyun.oss.Constants;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class OSSCredentialProvider implements CredentialsProvider {
```

Comment:
it's a little odd to use a mix of static credentials and token credential in this class, as if the fileset is using static credenital, no need to refresh.

Reply:
So how to use the static credential fetched from Gravitino, interface `CredentialsProvider` also supports static crendetial according to the definition. 

> as if the fileset is using static credenital, no need to refresh.

Based on the server-side expiration time, if it's 0 or Long.MAX, it will never refresh actually.  Do you have a better suggestion about it?

Reply:
How about getting credentials when initializing fileset, if it's a static credential, pass credential config items to the configuration, if it's a dynamic credential, pass `xxxCredentialProvider` and corresponding config items to the configuration.  `xxxCredentialProvider` is only used for refresh dynamic tokens.

Reply:
In my opinion for S3 and OSS `CredentialsProvider` should support dynamic and static credentials both according to the interface definition.  

> How about getting credentials when initializing fileset, if it's a static credential

I actually use this method for GCS and Azure. Please see the corresponding implementation. As we can merge dynamic and static credentials in the CredentialsProvider, I'm hesitant to split it into two for S3 and OSS.